lightkurve/injection.py

```diff
@@ -0,0 +1,288 @@
+        A parameter must be defined in the initialization of
+        TransitModel if it is to be changed in add_planet.
+
+        zpt : float
+            A photometric zeropoint
+        **kwargs : dict
+            Dictonary of planet parameters. Options are:
+                rho : stellar density
+                ld1, ld2, ld3, ld3 : limb darkening coefficients
+                dil : transit dilution fraction
+                veloffset : (not sure what this is)
+        """
+
+        self.zpt = zpt
+
+        self.star_params = {}
+        for key, value in kwargs.items():
+            if isinstance(value, (GaussianDistribution, UniformDistribution)):
+                self.star_params[key] = value.sample()
+            else:
+                self.star_params[key] = value
+
+        self.model.add_star(zpt=self.zpt, **self.star_params)
+
+    def add_planet(self, period, rprs, T0, **kwargs):
+        """Adds a planet to TransitModel object.
+
+        Parameters
+        ----------
+        period : float
+            Orbital period of new planet
+        rprs : float
+            Planet radius/star radius of new planet
+        T0 : float
+            A transit mid-time
+        **kwargs : dict
+            Dictonary of planet parameters. Options are:
+                impact: an impact parameter
+                ecosw, esinw : an eccentricity vector
+                occ : a secondary eclipse depth
+                rvamp : (not sure)
+                ell : (not sure)
+                alb : (not sure)
+        """
+
+        if isinstance(period, (GaussianDistribution, UniformDistribution)):
+            self.period = period.sample()
+        else:
+            self.period = period
+        if isinstance(rprs, (GaussianDistribution, UniformDistribution)):
+            self.rprs = rprs.sample()
+        else:
+            self.rprs = rprs
+        if isinstance(T0, (GaussianDistribution, UniformDistribution)):
+            self.T0 = T0.sample()
+        else:
+            self.T0 = T0
+
+        self.planet_params = {}
+        for key, value in kwargs.items():
+            if isinstance(value, (GaussianDistribution, UniformDistribution)):
+                self.planet_params[key] = value.sample()
+            else:
+                self.planet_params[key] = value
+
+        self.model.add_planet(period=self.period, rprs=self.rprs, T0=self.T0, **self.planet_params)
+
+    def evaluate(self, time):
+        """Creates lightcurve from model.
+
+        Parameters
+        ----------
+        time : array-like
+            Time array over which to create lightcurve
+            
+        Returns
+        _______
+        transit_flux : array-like
+            Flux array of lightcurve
+
+        """
+        self.model.add_data(time=time)
+        transit_flux = self.model.transitmodel
+        return transit_flux
+
+class SupernovaModel(object):
+
+    from lightkurve.injection import GaussianDistribution
```

Comment:
We should ask @barentsen how to do this properly.

We want to be able to use `isinstance(x, GaussianDistribution)` but it won't recognize `GaussianDistribution` as a class.